elements/howto-checkbox/howto-checkbox.js

```diff
@@ -0,0 +1,161 @@
+/**
+ * A `HowToCheckbox` represents a boolean option in a form. The most common type
+ * of checkbox is a dual-type which allows the user to toggle between two
+ * choices -- checked and unchecked.
+ */
+(function() {
+  /**
+   * Define key codes to help with handling keyboard events.
+   */
+  const KEYCODE = {
+    SPACE: 32,
+  };
+
+  /**
+   * The `HowToCheckbox` exposes a single `checked` attribute/property for
+   * toggling its state. Changes to the `checked` property will also be
+   * reflected to an `aria-checked` attribute. Similarly, the `disabled`
+   * property is reflected to an `aria-disabled` attribute. This controls
+   * whether the element is operable or not.
+   */
+  class HowToCheckbox extends HTMLElement {
+    static get observedAttributes() {
+      return ['checked', 'disabled'];
+    }
+
+    /**
+     * `connectedCallback` sets the initial `role`, `tabindex`,
+     * internal state, and installs event listeners.
+     */
+    connectedCallback() {
+      if (!this.hasAttribute('role'))
+        this.setAttribute('role', 'checkbox');
+      if (!this.hasAttribute('tabindex'))
+        this.setAttribute('tabindex', 0);
+
+      this._checked = false;
+      this._disabled = false;
+
+      this.addEventListener('keydown', this._onKeyDown);
+      this.addEventListener('click', this._onClick);
+    }
+
+    _onKeyDown(event) {
+      // Don’t handle modifier shortcuts typically used by assistive technology.
+      if (event.altKey)
+        return;
+
+      switch (event.keyCode) {
+        case KEYCODE.SPACE:
+          this._toggleChecked();
+          break;
+        // Any other key press is ignored and passed back to the browser.
+        default:
+          return;
+      }
+    }
+
+    _onClick(event) {
+      this._toggleChecked();
+    }
+
+    /**
+     * `_toggleChecked` calls the checked setter and flips its state.
+     * Because `_toggleChecked` is only caused by a user action, it will
+     * also dispatch a change event.
+     */
+    _toggleChecked() {
+      if (this.disabled)
+        return;
+      this.checked = !this.checked;
+      this.dispatchEvent(new CustomEvent('change', {
+        detail: {
+          checked: this.checked,
+        },
+        bubbles: false,
```

Comment:
My opinion on this comes from talking with Seb on the React team. He's of the opinion that elements should not bubble events, but instead their events should be handled by an immediate parent who then re-dispatches a more semantic event. I kinda like this notion so that's why I went with it here. I can add a comment explaining that and mention that it's a stylistic preference.